### PR TITLE
Expose patched method/function name in aspects API

### PR DESCRIPTION
**Pull request by seabaylea**
This provides the "methodName" that's been patched to the function callbacks to the aspects.before(), aspects.after() and aspects.around() calls so that its possible to know which function is being patched if an array of function names was provided.

Its passed as the new last parameter to avoid affecting existing code that uses the aspects calls.

### CODE REVIEW

**Comment by seabaylea** [comment]
@tunniclm Can you review this as the fix for issue #46

File: lib/aspect.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ exports.aroundCallback = function(args, hookBefore, hookAfter) {
 		var ret = orig.apply(this, arguments);
 
 		if(hookAfter) {
-			hookAfter(this, arguments);
+			hookAfter(this, arguments, ret);
```
Comment (tunniclm) [inline lib/aspect.js]: @seabaylea Is this change another fix? It doesn't look related to the method name change, unless I'm missing something?

**Comment by seabaylea** [comment]
Yeah, its a "small fix" which didn't look like it really deserved its own issue... hookAfter should have three parameters: this, arguments and return value. For some reason that call was missing the return value.

**Comment by tunniclm** [comment]
OK, LGTM then. Are you ready for it to be merged?

**Comment by seabaylea** [comment]
Yup - it should be good to go.
